amounts.go

```diff
@@ -4,19 +4,60 @@ import (
 	"fmt"
 	"math/big"
 	"regexp"
+	"strconv"
 
 	"github.com/pkg/errors"
-	samount "github.com/stellar/go/amount"
+	"github.com/stellar/go/xdr"
 )
 
-// Alow "-1", "1.", ".1", "1.1".
-// But not "." or "".
-var decimalStrictRE = regexp.MustCompile(`^-?((\d+\.?\d*)|(\d*\.?\d+))$`)
+// One is the value of one whole unit of currency. Stellar uses 7 fixed digits
+// for fractional values, thus One is 10 million (10^7).
+const (
+	One = 10000000
+)
+
+var (
+	bigOne = big.NewRat(One, 1)
+	// Alow "-1", "1.", ".1", "1.1".
+	// But not "." or "" or fractions or exponents
+	decimalStrictRE = regexp.MustCompile(`^-?((\d+\.?\d*)|(\d*\.?\d+))$`)
+)
 
-// ParseDecimalStrict parses a decimal number into a big rational.
+// ParseStellarAmount parses a decimal number into an int64 suitable
+// for the stellar protocol (7 significant digits).
+// See also stellar/go/amount#ParseInt64
+func ParseStellarAmount(v string) (int64, error) {
```

Comment:
worth it to drop that dependency? This package depends on that repo anyway. Seems like we'd never want to allow stuff that would fail their lib. They have tests. Need tests for this. Could copy their tests.

Reply:
i debated this and ultimately decided i thought it made the most sense to have the functions we're using to parse amounts and strings be in the same place since we often call different ones together. we could also push it down into `stellar-org/go/amounts.go` but that has its own weirdness. i'll take a gander at the tests, because i agree. this seems like something we should secure with tests.

Reply:
I have no qualms with the interface here. My concern is with the decision of copying in stellar/go/amount#ParseInt64 vs calling it. That decision needn't change the interface of this package.

Reply:
ahh. so yeah. i opted to use ParseAmount (was ParseDecimalStrict) first so i could reuse some of the extra checking. but it's just as easy to pull that out and call it from each. i'll take a look at this. there's some security and simplicity in using the method on their codebase.